homeassistant/components/shelly/update.py

```diff
@@ -0,0 +1,215 @@
+        entity_registry_enabled_default=True,
+    ),
+    "fwupdate_beta": RpcUpdateDescription(
+        name="Beta Firmware Update",
+        key="sys",
+        sub_key="available_updates",
+        installed_version=lambda status, shelly: shelly["ver"]
+        if shelly is not None
+        else None,
+        latest_version=lambda status, shelly: status.get("beta", {"version": None})[
+            "version"
+        ],
+        install=lambda wrapper: wrapper.async_trigger_ota_update(beta=True),
+        device_class=UpdateDeviceClass.FIRMWARE,
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up update entities for Shelly component."""
+    if get_device_entry_gen(config_entry) == 2:
+        return async_setup_entry_rpc(
+            hass, config_entry, async_add_entities, RPC_UPDATES, RpcUpdateEntity
+        )
+
+    if not config_entry.data[CONF_SLEEP_PERIOD]:
+        async_setup_entry_rest(
+            hass,
+            config_entry,
+            async_add_entities,
+            REST_UPDATES,
+            RestUpdateEntity,
+        )
+
+
+class RestUpdateEntity(ShellyRestAttributeEntity, UpdateEntity):
+    """Represent a REST update entity."""
+
+    _attr_supported_features = UpdateEntityFeature.INSTALL
+    entity_description: RestUpdateDescription
+
+    @property
+    def installed_version(self) -> str | None:
+        """Version currently in use."""
+        return str(
+            self.entity_description.installed_version(
+                self.wrapper.device.status,
+                self.wrapper.device.shelly,
+            )
+        )
+
+    @property
+    def latest_version(self) -> str | None:
+        """Latest version available for install."""
+        new_version = self.entity_description.latest_version(
+            self.wrapper.device.status,
+            self.wrapper.device.shelly,
+        )
+        if new_version is not None:
+            return str(new_version)
+
+        return self.installed_version
+
+    async def async_install(
+        self, version: str | None, backup: bool, **kwargs: Any
+    ) -> None:
+        """Install the latest firmware version."""
+        config_entry = self.wrapper.entry
+        block_wrapper = self.hass.data[DOMAIN][DATA_CONFIG_ENTRY][
+            config_entry.entry_id
+        ].get(BLOCK)
+        await self.entity_description.install(block_wrapper)
+
+
+class RpcUpdateEntity(ShellyRpcAttributeEntity, UpdateEntity):
+    """Represent a RPC update entity."""
+
+    _attr_supported_features = UpdateEntityFeature.INSTALL
+    entity_description: RpcUpdateDescription
+
+    @property
+    def installed_version(self) -> str | None:
+        """Version currently in use."""
+        return str(
+            self.entity_description.installed_version(
```

Comment:
Same as for block device, can be hardcoded and remove `shelly`

Reply:
done